accounts/abi/unpack.go

```diff
@@ -139,9 +139,9 @@ func forEachUnpack(t Type, output []byte, start, size int) (interface{}, error)
 	return refSlice.Interface(), nil
 }
 
-// toGoType parses the output bytes and recursively assigns the value of these bytes
+// ToGoType parses the output bytes and recursively assigns the value of these bytes
 // into a go type with accordance with the ABI spec.
-func toGoType(index int, t Type, output []byte) (interface{}, error) {
+func ToGoType(index int, t Type, output []byte) (interface{}, error) {
```

Comment:
I'm not happy about this change because you should be able to use

```go
var v interface{}
abi.Unpack(&v, "method", calldata)
```

Reply:
I'd be glad for some suggestions on how to do it 'properly' -- I did it the only way I saw possible. See https://github.com/holiman/go-ethereum/blob/signer_mhs/cmd/signer/abihelper.go#L65 for more context. 

I didn't know you could decode into an `interface{}`, I thought you had to allocate a suitable struct beforehand. I'll try your suggestion.

Reply:
That doesn't quite work, the `Unpack` function aims at decoding returndata, not inputs. 
It `panics` at e.g. 
```
	marshalledValue, err := ToGoType(0, method.Outputs[0].Type, output)
```
Because I'm throwing it input, and have not even defined any output. A hacky way to go about it would be to retranslate inputs to outputs in my abi, so that instead of supplying
```
{"type":"function","name":"send","inputs":[{"type":"uint256"}]}
```
I would supply
```
{"type":"function","name":"send","outputs":[{"type":"uint256"}]}
```

But is that nice?

Reply:
It's possible to make `Unpack` a bit more generic and handle inputs aswell as outputs... But I'm having some problems decoding 'tuple' types, though.

Reply:
IMHO we should make it work instead of exposing internals like `toGoType`.

Reply:
I agree. Let's try to clean up in the abi-swamp